File: backend/src/lib/dia-semana.ts
import { DiaSemana } from "@prisma/client";

export const DIAS_SEMANA = [
  { valor: "DOMINGO" as const, rotulo: "Domingo" },
  { valor: "SEGUNDA_FEIRA" as const, rotulo: "Segunda-feira" },
  { valor: "TERCA_FEIRA" as const, rotulo: "Terça-feira" },
  { valor: "QUARTA_FEIRA" as const, rotulo: "Quarta-feira" },
  { valor: "QUINTA_FEIRA" as const, rotulo: "Quinta-feira" },
  { valor: "SEXTA_FEIRA" as const, rotulo: "Sexta-feira" },
  { valor: "SABADO" as const, rotulo: "Sábado" },
] as const;

export const DIA_SEMANA_VALUES = DIAS_SEMANA.map((d) => d.valor) as [
  DiaSemana,
  ...DiaSemana[],
];

const porValor = new Map(DIAS_SEMANA.map((d) => [d.valor, d]));

export function rotuloDiaSemana(valor: DiaSemana): string {
  return porValor.get(valor)?.rotulo ?? valor;
}

File: mobile/lib/constants/dia_semana.dart
/// Dia da semana (valores da API / enum Prisma).
abstract final class DiaSemana {
  static const domingo = 'DOMINGO';
  static const segundaFeira = 'SEGUNDA_FEIRA';
  static const tercaFeira = 'TERCA_FEIRA';
  static const quartaFeira = 'QUARTA_FEIRA';
  static const quintaFeira = 'QUINTA_FEIRA';
  static const sextaFeira = 'SEXTA_FEIRA';
  static const sabado = 'SABADO';

  static const opcoes = <String, String>{
    domingo: 'Domingo',
    segundaFeira: 'Segunda-feira',
    tercaFeira: 'Terça-feira',
    quartaFeira: 'Quarta-feira',
    quintaFeira: 'Quinta-feira',
    sextaFeira: 'Sexta-feira',
    sabado: 'Sábado',
  };

  static String? rotulo(String? valor) {
    if (valor == null || valor.isEmpty) return null;
    return opcoes[valor];
  }
}
